src/tokenizer.rs

```diff
@@ -617,6 +618,36 @@ impl<'a> Tokenizer<'a> {
                         return Ok(Some(Token::Period));
                     }
 
+                    // Parse exponent as number
+                    if chars.peek() == Some(&'e') || chars.peek() == Some(&'E') {
+                        let mut char_clone = chars.peekable.clone();
```

Comment:
Why is this copy needed?

Given `chars` is already peekable I don't see why it can't be used directly

Reply:
I needed a way to peek more than just the next char, since is only valid exponent if `e` followed by optional sign and an actual number. Found easiest way was to simply clone the iter and use that, and if found not to be an exponent and safely discard it and continue regular behaviour with original iter.